src/openbiolink/graph_creation/graph_writer/bel_writer.py

```diff
@@ -0,0 +1,145 @@
+"""A utility for outputting graphs as BEL documents.
+
+To test, run ``openbiolink generate --no-download --no-input --output-format BEL --qual hq``.
+"""
+import logging
+import os
+from typing import Mapping, Optional, Set
+
+from tqdm import tqdm
+
+from openbiolink import Edge, EdgeType, Node, NodeType
+from openbiolink.graph_creation.graph_writer.base import GraphWriter
+
+__all__ = [
+    "GraphBELWriter",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class GraphBELWriter(GraphWriter):
+
+    def __init__(self, directory: Optional[str] = None, nodelink_gz: bool = True):
+        super().__init__(directory=directory)
+        self.nodelink_gz = nodelink_gz
+
+    def write(
+        self,
+        *,
+        tp_nodes: Mapping[str, Set[Node]],
+        tp_edges: Mapping[str, Set[Edge]],
+        tn_nodes: Mapping[str, Set[Node]],
+        tn_edges: Mapping[str, Set[Edge]],
+        **kwargs,
+    ) -> None:
+        """Write the graph as gzipped BEL graphs."""
+        from pybel import to_nodelink_gz
+
+        for nodes, edges, name in ((tp_nodes, tp_edges, "positive"), (tn_nodes, tn_edges, "negative")):
+            graph = convert(nodes=nodes, edges=edges, name=name)
+            graph_path = os.path.join(self.graph_dir_path, f"{name}.bel.nodelink.json.gz")
+            to_nodelink_gz(graph, graph_path)
+
+
+def _get_type_to_dsl():
+    import pybel.dsl
+
+    return {
+        NodeType.ANATOMY: pybel.dsl.Population,
+        NodeType.GENE: pybel.dsl.Protein,
+        NodeType.GO: pybel.dsl.BiologicalProcess,
+        NodeType.DIS: pybel.dsl.Pathology,
+        NodeType.DRUG: pybel.dsl.Abundance,
+        NodeType.PHENOTYPE: pybel.dsl.Pathology,
+        NodeType.PATHWAY: pybel.dsl.BiologicalProcess,
+    }
+
+
+def _get_type_to_adder():
+    from pybel import BELGraph
+
+    _type_to_adder = {
+        EdgeType.DRUG_BINDINH_GENE: BELGraph.add_directly_inhibits,
+        EdgeType.GENE_GENE: BELGraph.add_association,
+        EdgeType.GENE_OVEREXPRESSED_ANATOMY: BELGraph.add_positive_correlation,
+        EdgeType.GENE_GO: BELGraph.add_part_of,
+        EdgeType.GENE_INHIBITION_GENE: BELGraph.add_inhibits,
+        EdgeType.GENE_EXPRESSED_ANATOMY: BELGraph.add_association,
+        EdgeType.DRUG_ACTIVATION_GENE: BELGraph.add_activates,
+        EdgeType.DRUG_PREDBIND_GENE: None,
```

Comment:
Several of these edge types have a `None` because I'm not sure about the biological interpretation. could you explain what prebind is, what PTM the `GENE_PTMOD_GENE` means, what `GENE_REACTION_GENE` means, and what the catalysis interaction is?

Reply:
"Predbind" means "predicted to bind" and "ptmod" for "post-translational modifications" as in 4. of https://github.com/OpenBioLink/OpenBioLink/issues/22#issuecomment-582174838.
@matthias-samwald @annabreit Maybe you can help with this?